File: LeetCode/Simulation/Golang/Leetcode728-自除数.go
package sbgo

func selfDividingNumbers(left int, right int) []int {
	res := []int{}
	check := func(num int) bool {
		tmp := num
		for tmp != 0 {
			t := tmp % 10
			tmp /= 10
			if t == 0 || num%t != 0 {
				return false
			}
		}
		return true
	}
	for i := left; i <= right; i++ {
		if check(i) {
			res = append(res, i)
		}
	}
	return res
}

File: LeetCode/Simulation/Python/Leetcode728-自除数.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        res = []

        def check(num):
            tmp = num
            while tmp != 0:
                t = tmp % 10
                tmp //= 10
                if t == 0 or num % t != 0: return False
            return True

        for i in range(left, right + 1):
            if check(i): res.append(i)
        return res
